### PR TITLE
sql2html: support generation of schema diagrams as SVGs

### PR DESCRIPTION
**Pull request by mkszuba**
A vector format like SVG is generally more useful for this sort of data because it allows zooming in and out without loss of readability. This is important given how large some of our schema diagrams are. SVG files are also more space-efficient thank PNG ones, at least for the sort of resolutions GraphViz uses to produce the latter to make them readable anyway. On the other hand, the advantage of SVG over _e.g._ PDF is that it is natively supported as an image format by most Web browser _and_ can be directly exported to by GraphViz.

The format of diagram is controlled by a command-line option so it *is* still possible to generate PNGs should one choose to do so.

Please note that this PR introduces a dependency on ImageMagick (or GraphicsMagick with compatibility symlinks) because it uses the command 'identify' to extract dimensions from newly generated diagram files. Previously 'file' was used for that but it doesn't show dimensions of SVG files, moreover its output syntax varies wildly between formats (compare _e.g._ PNG and JPEG). This shouldn't be a big deal though, ImageMagick is fairly widespread (for one thing we have got it on the EBI farm) and most of its own dependencies are the same as those of GraphViz. Finally, if this PR gets accepted I would like to rewrite dimension extraction to use ImageMagick Perl bindings instead of an external executable - in which case we will be able to add this dependency to cpanfile.

### CODE REVIEW

**Comment by thomasmaurel** [comment]
@mkszuba: Thanks for the pull request. This script is in our repo but we don't run it and know nothing about it so I will ask @muffato and @ens-lgil to have a look at it.

**Comment by ens-lgil** [comment]
I believe this PR is more for @muffato to review it: I didn't wrote the part regarding the generation of the diagrams.

**Comment by mkszuba** [comment]
I would say let us err on the side of caution and choose the former for now, just in case. In theory SVG support in browsers, or at least the parts relevant to such simple graphics as our schema diagrams, should be mature by now - but then again who knows what it is really like, or what our users actually use.

I've just deleted the commit which changed the default from PNG to SVG.

File: scripts/sql2html.pl [hunk 1]
```diff
@@ -1784,6 +1813,7 @@ sub usage {
     -sort_headers     A flag to sort (1) or not (0) the headers by alphabetic order. By default, the value is set to 1.
     -sort_tables      A flag to sort (1) or not (0) the tables by alphabetic order. By default, the value is set to 1.
     'out_diagram_dir  Where to generate schema diagrams (one for the whole schema, and one per category). By default, the value is not set, meaning no diagrams are generated.
+    -diagram_format   File format to be used for generated schema diagrams. Supported values: png, svg (default).
```
[the remark this answers is not in the document]
Comment (mkszuba) [reply scripts/sql2html.pl]: Whoops, that had got caught into the wrong commit. Fixed.

**Pull request merged**